ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,407 @@
+        check_file_path(
+            self.parameter, self.parameter_name, self.expected_file_extensions
+        )
+
+
+class OptionalFilePathCheck(BaseInputArgumentChecker):
+    """Class to check optional file_path-like parameters"""
+
+    def __init__(self, parameter, parameter_name, expected_file_extension):
+        self.parameter = parameter
+        self.parameter_name = parameter_name
+        self.expected_file_extensions = expected_file_extension
+
+    def check(self):
+        """Method raises ValueError exception if file path parameter is not equal to expected"""
+        if self.parameter is not None:
+            check_file_path(
+                self.parameter, self.parameter_name, self.expected_file_extensions
+            )
+
+
+class DatasetParamTypeCheck(BaseInputArgumentChecker):
+    """Class to check DatasetEntity-type parameters"""
+
+    def __init__(self, parameter, parameter_name):
+        self.parameter = parameter
+        self.parameter_name = parameter_name
+
+    def check(self):
+        """Method raises ValueError exception if parameter is not equal to DataSet"""
+        check_is_parameter_like_dataset(
+            parameter=self.parameter, parameter_name=self.parameter_name
+        )
+
+
+class OptionalDatasetParamTypeCheck(DatasetParamTypeCheck):
+    """Class to check DatasetEntity-type parameters"""
+
+    def check(self):
+        """Method raises ValueError exception if parameter is not equal to DataSet"""
+        if self.parameter is not None:
+            check_is_parameter_like_dataset(
+                parameter=self.parameter, parameter_name=self.parameter_name
+            )
+
+
+class OptionalModelParamTypeCheck(BaseInputArgumentChecker):
+    """Class to check ModelEntity-type parameters"""
+
+    def __init__(self, parameter, parameter_name):
+        self.parameter = parameter
+        self.parameter_name = parameter_name
+
+    def check(self):
+        """Method raises ValueError exception if parameter is not equal to DataSet"""
+        if self.parameter is not None:
+            for expected_attribute in (
+                "__train_dataset__",
+                "__previous_trained_revision__",
+                "__model_format__",
+            ):
+                if not hasattr(self.parameter, expected_attribute):
+                    parameter_type = type(self.parameter)
+                    raise ValueError(
+                        f"parameter '{self.parameter_name}' is not like ModelEntity, actual type: {parameter_type} "
+                        f"which does not have expected '{expected_attribute}' Model attribute"
+                    )
+
+
+class OptionalImageFilePathCheck(OptionalFilePathCheck):
+    """Class to check optional image file path parameters"""
+
+    def __init__(self, parameter, parameter_name):
+        super().__init__(
+            parameter=parameter,
+            parameter_name=parameter_name,
+            expected_file_extension=["jpg", "png"],
```

Comment:
https://docs.opencv.org/3.4/d4/da8/group__imgcodecs.html#ga288b8b3da0892bd651fce07b3bbd3a56

Reply:
Thanks, I will update the list of expected files extensions according to the imread methods supported formats

Reply:
Updated expected types, but in specified link there are no extensions included for “Raster and Vector geospatial data supported by GDAL” should I add all extensions from https://svn.osgeo.org/gdal/tags/gdal_1_2_5/frmts/formats_list.html for Raster and for https://gdal.org/drivers/vector/index.html Vector?